public_budget/view/budget_modification_detail_view.xml

```diff
@@ -50,5 +57,22 @@
             </field>
         </record>
 
+        <!-- TREEVIEW -->
+        <record id="view_budget_modification_detail_tree_inherit" model="ir.ui.view">
```

Comment:
search view is searching and grouping by budget_position_id field, but tree view does not have this field, not even as a invisible one. I am not sure but maybe this could be broken. AFAIK Field should be in the view in order to do the .search Do you test if this search filters are working?